python/GafferSceneUI/ImageToPointsUI.py

```diff
@@ -0,0 +1,170 @@
+
+import Gaffer
+import GafferScene
+
+##########################################################################
+# Metadata
+##########################################################################
+
+Gaffer.Metadata.registerNode(
+
+	GafferScene.ImageToPoints,
+
+	"description",
+	"""
+	Converts an image into a 3d pointcloud, with a point for each pixel
+	in the image. Point positions may be defined either by the original
+	pixel coordinates or an image layer providing position data.
+	Arbitrary image channels may be converted to additional primitive
+	variables on the pointcloud, and transparent pixels may be omitted
+	from the conversion.
+
+	> Note : Only pixels within the display window are converted. To
+	> include overscan pixels, use a Crop node to extend the display
+	> window.
+	""",
+
+	plugs = {
+
+		"sets" : [
+
+			"layout:divider", True,
+
+		],
+
+		"image" : [
+
+			"description",
+			"""
+			The image to be converted into a pointcloud.
+			""",
+
+			"nodule:type", "GafferUI::StandardNodule",
+
+		],
+
+		"view" : [
+
+			"description",
+			"""
+			The view within the image to be converted.
+			""",
+
+			"plugValueWidget:type", "GafferImageUI.ViewPlugValueWidget",
+			"layout:divider", True,
+
+		],
+
+		"position" : [
+
+			"description",
+			"""
+			The image channels used to provide 3d positions for the points.
+			If `None`, the pixel's 2d position within the image is used instead.
+			""",
+
+			"plugValueWidget:type", "GafferImageUI.RGBAChannelsPlugValueWidget",
+			"rgbaChannelsPlugValueWidget:allowNone", True,
+
+		],
+
+		"primitiveVariables" : [
+
+			"description",
+			"""
+			The image channels to be converted to primitive variables on
+			the pointcloud. The chosen channels are converted using the
+			following rules :
+
+			- The main `RGB` channels are converted to a colour primitive variable called `Cs`.
+			- `<layerName>.RGB` channels are converted to a colour primitive variable called `<layerName>`.
+			- Other channels are converted to individual float primitive variables.
+			""",
+
+			"plugValueWidget:type", "GafferImageUI.ChannelMaskPlugValueWidget",
+
+		],
+
+		"width" : [
+
+			"description",
+			"""
+			The width of the points. If `widthChannel` is used as well, then this acts as
+			a multiplier on the channel values.
+			"""
+
+		],
+
+		"widthChannel" : [
+
+			"description",
+			"""
+			The channel used to provide per-point width values for the pointcloud.
+			""",
+
+			"plugValueWidget:type", "GafferImageUI.ChannelPlugValueWidget",
+			"channelPlugValueWidget:imagePlugName", "image",
+			"layout:divider", True,
+
+		],
+
+		"ignoreTransparent" : [
+
+			"description",
+			"""
+			Omits pixels from the pointcloud if their alpha value is less than or equal
+			to `alphaThreshold`.
+			""",
+
+		],
+
+		"alphaThreshold" : [
+
+			"description",
+			"""
+			Threshold used to exclude pixels from the pointcloude when `ignoreTransparent`
```

Comment:
Ye olde pointcloude shoppe.

Reply:
Also fixed in https://github.com/GafferHQ/gaffer/pull/5177/commits/a771ceb6acd146ad66ab1336f7969510a03a75c3.

Maybe this is how we should encourage long-running shows to be a bit more ambitious in their upgrade schedules - regress the language in the documentation gradually back towards the 16th century with each point release :)